python/mleap/pyspark/feature/string_map.py

```diff
@@ -26,14 +47,3 @@ def __init__(self, labels, inputCol=None, outputCol=None, handleInvalid='error',
         self._java_obj = self._new_java_obj("org.apache.spark.ml.mleap.feature.StringMap", self.uid, string_map_model)
         self.setInputCol(inputCol)
         self.setOutputCol(outputCol)
-
-    @classmethod
-    def from_dataframe(cls, df, key_col, value_col, handleInvalid='error', defaultValue=0.0):
```

Comment:
This is cleaner, because there's no need to echo the same kwargs as required by the constructor.

I also wrote this in PR desc about it:

> Remove separate from_dataframe method. Allow passing a DataFrame as an alternative to a dict in the same constructor.

Reply:
Repeating args is not a big problem. I'd argue that the @classmethod is a more pythonic way to handle multiple different constructor. 
This is an example of the multiple constructor pattern via @classmethod: https://stackoverflow.com/a/682545

A problem is that `labels` becomes a variable holding objects of different types (generally not even allowed in other languages) that is python strength and curse at the same time.

Out unique constructor includes now an if statement  (`if isinstance(labels, DataFrame):`) breaking the [Principle of Single Responsibility](https://en.wikipedia.org/wiki/Single_responsibility_principle) for our __init__ function. We also get a hidden behaviour (a special handling of Dataframes) in the __init__ that goes against transparency.

Python doesn't have typed values so sometimes we must be explicit with naming. Take this example:
```python
some_data = DictOrDataframe()

# reader doesn know if some_data is a dict or a dataframe
StringMap(some_data)

# reader knows some_data is a dataframe
StringMap.from_dataframe(some_data)
```